jetstream/config.py

```diff
@@ -62,64 +68,91 @@
+    """
+    Takes platform configuration and generates platform object map
+    """
+
+    valid_modules_metrics = mozanalysis.metrics.__all__
+    valid_modules_segments = mozanalysis.segments.__all__
+
+    _valid_entrollments_query_types = (
         "glean-event",
-        "org.mozilla.ios.Focus",
-    ),
-    "klar_ios": Platform(
-        Path(__file__).parent / "config" / "klar_ios.toml",
-        mozanalysis.metrics.klar_ios,
+        "normandy",
+    )
+
+    _none_values = [
+        "none",
+        "None",
         None,
-        "glean-event",
-        "org.mozilla.ios.Klar",
-    ),
-}
+    ]
+
+    processed_config = dict()
+    _errors = list()
+
+    for platform, platform_config in config["platform"].items():
+        if metrics_module := platform_config.get("metrics_module"):
+            if metrics_module in _none_values: metrics_module = None
+            if metrics_module not in [*valid_modules_metrics, *_none_values]:
+                raise Exception()  # TODO: fix up
+
+        if segments_module := platform_config.get("segments_module"):
+            if segments_module in _none_values: segments_module = None
+            if segments_module not in [*valid_modules_segments, *_none_values]:
+                raise Exception()  # TODO: fix up
+
+        enrollments_query_type = platform_config["enrollments_query_type"]
+        if enrollments_query_type not in _valid_entrollments_query_types:
+            raise AttributeError(
+                "%s is not a valid selection. Valid options are: %s. \
+                    If this is correct please add %s to _valid_entrollments_query_types"
+                % (
+                    enrollments_query_type,
+                    _valid_entrollments_query_types,
+                    _valid_entrollments_query_types,
+                )
+            )
+
+        try:
+            processed_config[platform] = {
+                "config_spec_path": CONFIG_DIRECTORY / platform_config["config_spec"],
+                "metrics_module": importlib.import_module(f"mozanalysis.metrics.{metrics_module}") if metrics_module else None,
+                "segments_module":importlib.import_module(f"mozanalysis.segments.{segments_module}")  if segments_module else None,
+                "enrollments_query_type": enrollments_query_type,
+                "validation_app_id": platform_config["validation_app_id"],
+            }
+        except KeyError as _err:
+            _errors.append(_err)
+            logging.error(
+                "%s is not a valid selection. Please ensure that this \
+                    metric or segment have a mapping defined inside \
+                    _module_mapping_metrics or _module_mapping_segment"
+                % (_err)
+            )
+
+    if _errors:
+        raise AttributeError(
+            "Please ensure that the metric(s) and/or segment(s) have \
+                a mapping defined inside _module_mapping_metrics and/or _module_mapping_segment"
+        )
```

Comment:
I guess we can do some cleanup here with the new changes.

Reply:
Yes, I will do that today.